node/building_blocks.re

```diff
@@ -1,9 +1,7 @@
 open Helpers;
 open Protocol;
 
-module Node = State;
-
-let is_valid_block = (state, block) => {
+let is_valid_block = (state: Node_state.t, block) => {
```

Comment:
Never add unneeded type annotation

Reply:
I did not intend to add unneeded type annotations, only those that were needed. I could easily have made a mistake in this respect. 

In general, do you prefer: `x.ModuleA.propA` over `let x : ModuleA = ... in x.propA`?